### PR TITLE
Upgrade to Tendermint 0.9.2

### PR DESCRIPTION
**Pull request by silasdavis**
This PR:

- Makes changes necessary to support Tendermint 0.9.2 and upgrades our dependencies to match
- Adds attempt to gracefully shutdown ConsensusEngine
- Updates RPC tests to use genesis making and config generation rather than hard-coded values

### CODE REVIEW

File: core/core.go [hunk 1]
```diff
@@ -119,3 +119,8 @@ func (core *Core) NewGatewayTendermint(config *server.ServerConfig) (
 	return rpc_tendermint.NewTendermintWebsocketServer(config,
 		core.tendermintPipe, core.evsw)
 }
+
+// Stop the core allowing for a graceful shutdown of component in order.
+func (core *Core) Stop() bool {
```
Comment (benjaminbollen) [inline core/core.go]: Nice

File: manager/burrow-mint/burrow-mint.go [hunk 2]
```diff
@@ -177,6 +186,32 @@ func (app *BurrowMint) Commit() (res abci.Result) {
...
+// Signals the beginning of a block
+func (app *BurrowMint) BeginBlock(hash []byte, header *abci.Header) {
+
 }
+
+// Signals the end of a blockchain, return value can be used to modify validator
+// set and voting power distribution see our BlockchainAware interface
+func (app *BurrowMint) EndBlock(height uint64) (respEndblock abci.ResponseEndBlock) {
+	// TODO: [Silas] Bondage
+	// TODO: [Silas] this might be a better place for us to dispatch new block
```
Comment (benjaminbollen) [inline manager/burrow-mint/burrow-mint.go]: Agreed.

Comment (silasdavis) [inline manager/burrow-mint/burrow-mint.go]: Actually only issue with that is is that I think they are sending ABCI EndBlock _before_ they save Tendermint state. Not an issue for something things, but may be if you were expecting post-block consensus state to be ready.

**Comment by benjaminbollen** [comment]
closes https://github.com/hyperledger/burrow/issues/583

**Pull request merged**